lib/glimmer/tk/widget_proxy.rb

```diff
@@ -551,21 +562,91 @@ def handle_listener(listener_name, &listener)
       def on(listener_name, &listener)
         handle_listener(listener_name, &listener)
       end
-      
+
+      def raise_event(event_name, data = nil)
+        data = YAML.dump(data) if data && !data.is_a?(String)
+        tk.event_generate("<#{event_name}>", data: data)
+      end
+
       def unbind_all
         @listeners&.keys&.each do |key|
           if key.to_s.downcase.include?('command')
             @tk.send(key, '')
           else
-            @tk.bind(key, '')
+            @tk.bind_remove(key)
           end
         end
+        @listeners = nil
       end
-      
+
+      def clear
```

Comment:
`WidgetProxy#clear` is too vague of a method name. For example, calling `clear` on a `label` makes no sense in this case as it gives the impression that we want to clear the label text. 

Please rename to `clear_children`